### ISSUE TITLE
RPN 

### ISSUE THREAD

**Issue by Nazarko12345** [learnml-today/object-detection-with-pytorch#18]
- Why we detach output of rpn for box_deltas?If we wont detach outputs we will get strong region proposals and then more stonger coordinates of box i was thinking that its main idea of faster-rcnn like double converging on coordinates.If we always generating random boxes what the point of it?Or it will converge but very slowly because of gradient descent on confidence output?
 

- Why we get final output :
`self.cls_score = nn.Linear(in_channels, num_classes) # (N_boxes, num_classes)
  self.bbox_pred = nn.Linear(in_channels,  num_classes*4) # (N_boxes, num_classes*4) bbox for each class ` 
In final output we dont use anchors we predict just coordinates?And why for each class not just for certain class?

**Comment by creotiv** [comment]
1) That was because ROI pool/align was not differentiable, but i just looked and seems it was fixed. So you can try to do joint training and see if it improves

2) don't quite understand about anchor boxes, we always predict coordinates. 
as for bbox for all classes it just forcing model to not compete on class bases, thus training simpler
